graphqlapi/src/main/java/io/stargate/graphql/schema/schemafirst/fetchers/admin/SchemaFetcher.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.graphql.schema.schemafirst.fetchers.admin;
+
+import graphql.schema.DataFetchingEnvironment;
+import io.stargate.auth.AuthenticationSubject;
+import io.stargate.auth.AuthorizationService;
+import io.stargate.auth.SourceAPI;
+import io.stargate.auth.UnauthorizedException;
+import io.stargate.db.datastore.DataStore;
+import io.stargate.db.datastore.DataStoreFactory;
+import io.stargate.graphql.persistence.schemafirst.SchemaSourceDao;
+import io.stargate.graphql.schema.CassandraFetcher;
+import java.util.Collections;
+
+public abstract class SchemaFetcher<ResultT> extends CassandraFetcher<ResultT> {
+  public SchemaFetcher(
+      AuthorizationService authorizationService, DataStoreFactory dataStoreFactory) {
+    super(authorizationService, dataStoreFactory);
+  }
+
+  protected void authorize(AuthenticationSubject authenticationSubject)
+      throws UnauthorizedException {
+    authorizationService.authorizeSchemaRead(
```

Comment:
Would it be possible someone would have access to the `schema_source` table but not the actual table they're getting the schema for? Just wondering if we should be authorizing the `stargate_graphql` keyspace and the namespace they're working with?

Reply:
Right, it would make sense to also check the keyspace here.